lib/model/model.go

```diff
@@ -1434,23 +1432,36 @@ func (m *model) Closed(conn protocol.Connection, err error) {
 	close(closed)
 }
 
-// closeConns will close the underlying connection for given devices
-func (m *model) closeConns(devs []protocol.DeviceID, err error) {
+// closeConns will close the underlying connection for given devices and
+// return an array of channels that will be closed once the connections are
+// finished closing.
+func (m *model) closeConns(devs []protocol.DeviceID, err error) []chan struct{} {
 	conns := make([]connections.Connection, 0, len(devs))
+	closed := make([]chan struct{}, 0, len(devs))
 	m.pmut.Lock()
 	for _, dev := range devs {
 		if conn, ok := m.conn[dev]; ok {
 			conns = append(conns, conn)
+			closed = append(closed, m.closed[dev])
 		}
 	}
 	m.pmut.Unlock()
 	for _, conn := range conns {
 		conn.Close(err)
 	}
+	return closed
```

Comment:
Not a fan of this []chan approach. We always wait for all of them, perhaps return a Waiter() or a wait group?

Reply:
Went with a `Waiter`, definitely neater that way, thanks.